src/main/java/com/matchbook/sdk/ClientConfig.java

```diff
@@ -0,0 +1,69 @@
+package com.matchbook.sdk;
+
+public class ClientConfig {
+
+    private String url;
+    private String loginUrl;
+    private char[] username;
+    private char[] password;
+
+    private ClientConfig() {
+    }
+
+    public static class Builder {
+        private String url;
+        private String loginUrl;
+        private char[] username;
+        private char[] password;
+
+        public Builder(char[] username, char[] password) {
+            this.username = username;
+            this.password = password;
+        }
+
+        public Builder username(char[] user) {
+            this.username = user;
+            return this;
+        }
+
+        public Builder password(char[] password) {
+            this.password = password;
+            return this;
+        }
+
+        public Builder url(String url) {
+            this.url = url;
+            return this;
+        }
+
+        public Builder loginUrl(String loginUrl) {
+            this.loginUrl = loginUrl;
+            return this;
+        }
+
+        public ClientConfig build() {
+            ClientConfig clientConfig = new ClientConfig();
```

Comment:
I would move the creation of the `ClientConfig` in the constructor.

For instance
```suggestion
            return new ClientConfig(this);
```
and then the private `ClientConfig` constructor
```suggestion
    private ClientConfig(ClientConfig.Builder builder) {
        this.url = builder.url;
        this.username = builder.username;
        this.password = builder.password;
        this.loginUrl = builder.loginUrl;
    }
```